ui/src/components/Inventories/InventoryFormModal.js

```diff
@@ -0,0 +1,189 @@
+import Button from '@material-ui/core/Button'
+import Dialog from '@material-ui/core/Dialog'
+import DialogActions from '@material-ui/core/DialogActions'
+import DialogContent from '@material-ui/core/DialogContent'
+import DialogTitle from '@material-ui/core/DialogTitle'
+import Grid from '@material-ui/core/Grid'
+import React from 'react'
+import TextField from '../Form/TextField'
+import { Field, Form, Formik } from 'formik'
+import { MeasurementUnits } from '../../constants/units'
+import { Checkbox, FormControlLabel, MenuItem } from '@material-ui/core'
+import products from '../../ducks/products'
+
+const measurementUnitKeys = Object.keys(MeasurementUnits)
+const allProducts= products
+const date = new Date()
+class InventoryFormModal extends React.Component {
+  
+  render() {
+    const {
+      formName,
+      handleDialog,
+      handleProduct,
+      title,
+      initialValues
+    } = this.props
+    return (
+      <Dialog
+        open={this.props.isDialogOpen}
+        maxWidth='sm'
+        fullWidth={true}
+        onClose={() => { handleDialog(false) }}
+      >
+        <Formik
+          initialValues={initialValues}
+          onSubmit={values => {
+            handleProduct(values)
+            handleDialog(true)
+          }}>
+          {helpers =>
+            <Form
+              noValidate
+              autoComplete='off'
+              id={formName}
+            >
+              <DialogTitle id='alert-dialog-title'>
+                {`${title} Inventory`}
+              </DialogTitle>
+              <DialogContent>
+                <Grid container>
+                  <Grid item xs={12} sm={12}>
+                    <Field
+                      custom={{ variant: 'outlined', fullWidth: true, }}
+                      name='name'
+                      label='Name'
+                      required
+                      component={TextField}
+                    />
+                  </Grid>
+                </Grid>
+                <Grid container>
+                  <Grid item xs={12} sm={12}>
+                  <Field
+                    select
+                    custom={{ variant: 'outlined', fullWidth: true, }}
+                    name='product type'
+                    label='Product Type'
+                    component={TextField}
+                    >
+                      {allProducts.map((option) => (
```

Comment:
All the logic is right. Rename `option` to `product` to keep naming consistent; `mapping` is essentially a loop through `products`.
`allProducts` should be replaced with the passed down `products`.